Demos 739 phase 3 - completeness!

dustbuster opened the pull request:
Phase 3, as wired up as i can make it without real dates or docs!

## Completeness Video
https://github.com/user-attachments/assets/f8277bcc-34a4-4444-8c5e-9865f5783a42



<img width="692" height="479" alt="Screenshot 2025-09-25 at 16 54 36" src="https://github.com/user-attachments/assets/58e686b4-cf4b-4733-844f-bd1412e2fc4b" />


<img width="730" height="558" alt="Screenshot 2025-09-25 at 16 54 45" src="https://github.com/user-attachments/assets/c754cd20-39d6-4df0-849f-addff8aebc75" />

dustbuster committed:
DEMOS-769: set up basic frame work to start in

Completeness page

dustbuster committed:
DEMOS-769: added notice and did some clean up

1. notice is like a toast that deos not time out.
2. Rearranged page to put testing panel at bottom

dustbuster committed:
Merge branch 'main' into DEMOS-769

dustbuster committed:
DEMOS-769: updated file to work with changing code

There was a person update and my code needed updating

dustbuster committed:
Merge branch 'main' into DEMOS-769

dustbuster committed:
DEMOS-769: cleaned up notice component, match spec

Re-usable component for semi-perm toasts.

dustbuster committed:
DEMOS-769: refactored some TW classes together

dustbuster committed:
DEMOS-739: added some logic to notice for due date

There is a notice box that turns red whne its geting close to due date.

dustbuster committed:
DEMOS-739: added the incomplete dialog.

This is just a box that captures the reason why incomplete.

dustbuster committed:
DEMOS-739: Added complete to the completionPhase

1. This is the other half of this.

dustbuster committed:
DEMOS-739: added tests and added markers to test

Data-testid's added and added tests for notices for Completeness phase

dustbuster committed:
DEMOS-739: pulled out test panel into its own comp

This is to make it portable.

dustbuster committed:
DEMOS-739: Added comment for context

dustbuster committed:
Merge branch 'main' into DEMOS-739-phase-3

dustbuster committed:
DEMOS-739: based screen effects based on real date

1. Before there was a "set remaining days" but this is based on the due
   date you drop in now.
2. Adjusted the color of the due date in phase block to match the toast
   black if rest.
3. Various cleanups

dustbuster committed:
DEMOS-739: updated some tests and addded test-id

to testing panel

dustbuster committed:
DEMOS-739: got the date working in all places

removed the default and we rely on testing panel.
Pretty much wired up all the data messery

dustbuster committed:
DEMOS-739: some minor style and default cleanup

dustbuster committed:
DEMOS-739: added todo for fake message

dustbuster committed:
Merge branch 'main' into DEMOS-739-phase-3

dustbuster committed:
DEMOS-739: fixed some linter issues

dustbuster commented on client/src/components/application/phases/CompletenessPhase.tsx:
```diff
@@ -1,5 +1,325 @@
+  const [federalStartDate, setFederalStartDate] = useState<string>("");
+  const [federalEndDate, setFederalEndDate] = useState<string>("");
+
+  const completenessDocs = mockDocuments.filter(
+    (doc) => doc.documentType === "Application Completeness Letter"
+  );
```
Easy wire in once we get the real doc in here.

dustbuster commented on client/src/components/application/phases/CompletenessPhase.tsx:
```diff
@@ -1,5 +1,325 @@
+        </div>
+      )}
+      {/* Since this is pretty handy (pre-api hookup), let's make it easy to separate */}
+      {isLocalDevelopment() && (
+        <CompletenessTestingPanel
+          onAddMockDoc={addMockDoc}
+          completenessDocCount={completenessDocs.length}
+          noticeDueDate={noticeDueDate}
+          onNoticeDueDateChange={setNoticeDueDate}
+          noticeDaysValue={noticeDaysValue} // keep for display-only if you show it
+          onResetNotice={() => setNoticeDismissed(false)}
+        />
```
basically this lil control panels tells us what we need to hydrate!

dustbuster committed:
DEMOS-739: added "move to next phase" on fimish

This way, you just automagically get moved to the next phase. and it's
supposed to nuke the notice.

dustbuster committed:
Merge branch 'main' into DEMOS-739-phase-3

dustbuster committed:
DEMOS-739: removed a couple commented out things

also updated the package-json

dustbuster committed:
DEMOS-739: reset the package jsons

tresdonicf commented on client/src/components/notice/Notice.tsx:
Awesome stuff!

tresdonicf commented on client/src/components/dialog/DeclareIncompleteDialog.tsx:
I kind of have a hunch that this should be bundled with the component it's used since it's not general enough to be re-used across different parts of the application

Looks great though!

tresdonicf commented on client/src/components/dialog/document/CompletenessUploadDialog.tsx:
```diff
@@ -0,0 +1,45 @@
+  isOpen: boolean;
+  onClose: () => void;
+  bundleId?: string; // optional starter prop; can be wired later
+  refetchQueries?: string[];
```
Let's maintain the list of queries to refetch rather than have the user pass them in

tresdonicf commented on client/src/components/dialog/document/CompletenessUploadDialog.tsx:
Same thing with this guy, starting to notice we might want to put some of these just right next to where they're used since it will only likely be used in one place

dustbuster replied:
Done.

tresdonicf commented on client/src/components/application/phase-selector/PhaseDate.tsx:
```diff
@@ -26,13 +28,24 @@ const PHASE_TEXT_COLOR_LOOKUP: Record<PhaseStatus, string> = {
+interface PhaseDateProps {
+  phaseStatus: PhaseStatus;
+  date?: Date;
+  isPastDue?: boolean;
```
For the purposes of this component, which is just for rendering the phase dates I think this should just be a new `PhaseStatus`  like `past-due` (or any name like that)

Then that status is calculated before passing it into this component

tresdonicf replied:
This means we can also re-use the existing lookups rather than handle it as a separate edge case which it isn't quite

dustbuster committed:
DEMOS-739: swapped out date stuff with date-fns

This allows for consistent date management and expandability.

tresdonicf commented on client/src/components/application/phase-selector/PhaseBox.tsx:
```diff
@@ -30,6 +30,7 @@ interface PhaseBoxProps {
   phaseStatus: PhaseStatus;
   displayDate?: Date;
+  isPastDue?: boolean;
   isSelectedPhase: boolean;
```
I think the way forward here is adding this to our `PhaseStatus` type. We could do it in the server and add a status for past-due or we could just add it on the frontend here:

`type PhaseStatus = ServerPhaseStatus | "past-due"`. If I remember right I think that there might be a need to represent past due on the server anyway for some type of requirement thing but I could be wrong, might be worth a conversation with the team

tresdonicf commented on client/src/components/application/phases/CompletenessPhase.tsx:
```diff
@@ -1,5 +1,325 @@
+  const [year, month, day] = input.split("-").map((value) => Number(value));
+  if (!year || !month || !day) return undefined;
+  return new Date(year, month - 1, day);
+};
```
Probably can use stuff from date-fns for these

tresdonicf commented on client/src/components/application/phase-selector/PhaseSelector.tsx:
I think this is the same as the PhaseNames in the server now, is that right? Oh except maybe `None` - if so perhaps we can use that now that it's ready

tresdonicf commented on client/src/components/application/phase-selector/PhaseSelector.tsx:
```diff
@@ -69,63 +71,152 @@ interface PhaseSelectorProps {
     return (
       <div className="w-full h-full min-h-64">
-        <PhaseComponent />
+        {PhaseComponent ? <PhaseComponent /> : null}
```
PhaseComponent should always exist here as we've made the lookup and are using the values from it.

dustbuster committed:
DEMOS-739: Set the notice to only show up if data

This way there's no mock messages we have to nuke

tresdonicf reviewed (dismissed):
Overall this is looking good! The one change that I think fragments the system a bit too  much is the handling of the `pastDue` status separately from the other `PhaseStatus`es. I think some of the tooling for `PhaseMeta` and stuff are built out of that but we really don't need anything like that as the whole `phases` model in the dataflow is pretty unstructured so we can put any dates we need into there.

So if we need to capture the due date for a phase we should make sure it exists in `DATE_TYPES` ("Completeness Review Due Date" is already there) and interact with that date if it exists, and setting the `PhaseStatus` based on that

dustbuster committed:
DEMOS-739: added phase status type and implemented.

1. Moved from an attribute to a tyoe for over due statyes
1 this fixes the issue with

dustbuster committed:
Merge branch 'main' into DEMOS-739-phase-3

dustbuster committed:
DEMOS-739: use the PHASE_NAMES from constants

dustbuster committed:
DEMOS-739: moved the doc dialog to better home

Since this will prolly not get sued again, let's just plop it nex to
where it's used

dustbuster committed:
DEMOS-739: simplified the phase selector page.

Phase metadata now only tracks the stored due date, and PhaseSelector derives the past_due status by comparing that date at render time; the state updater no longer shuffles extra flags form PhaseSelector and PhaseStatusContext.tsx

The completeness phase simply publishes its selected due date into the selector context, keeping the badge color logic entirely inside the selector

dustbuster committed:
DEMOS-739: updated the test to use date-fns

dustbuster committed:
DEMOS-739: removed phaseStatus, AI overkill

dustbuster commented:
okay. This has some useful stuff that might come in handy for the wiring. Will close and revisit if needed.
